src/services/mentees.js

```diff
@@ -781,7 +782,146 @@ module.exports = class MenteesHelper {
 	}
 
 	/**
-	 * List mentees and search with name , email
+	 * Get entities and organization filter
+	 * @method
+	 * @name getFilterList
+	 * @param {String} tokenInformation - token information
+	 * @param {Boolean} queryParams - queryParams
+	 * @returns {JSON} - Filter list.
+	 */
+	static async getFilterList(queryParams, tokenInformation) {
+		try {
+			let result = {
+				organizations: [],
+				entity_types: {},
+			}
+
+			let organization_ids = []
+			const organizations = await this.getOrganizationIdBasedOnPolicy(tokenInformation.id)
+			if (organizations.success && organizations.result.length > 0) {
+				organization_ids = [...organizations.result]
+
+				if (organization_ids.length > 0) {
+					//get organization list
+					const organizations = await userRequests.listOrganization(organization_ids)
+					if (organizations && organizations.result?.length > 0) {
+						result.organizations = organizations.result
+					}
+
+					const defaultOrgId = await getDefaultOrgId()
+					if (defaultOrgId && !organization_ids.includes(defaultOrgId)) {
+						organization_ids.push(defaultOrgId)
+					}
+
+					//get entity type with entities list
+					const getEntityTypesWithEntities = await this.getEntityTypeWithEntitiesBasedOnOrg(
+						organization_ids,
+						queryParams
+					)
+					if (getEntityTypesWithEntities.success && getEntityTypesWithEntities.result) {
+						let entityTypesWithEntities = getEntityTypesWithEntities.result
+						if (entityTypesWithEntities.length > 0) {
+							let convertedData = convertEntitiesForFilter(entityTypesWithEntities)
+							result.entity_types = filterEntitiesBasedOnParent(convertedData, defaultOrgId)
+						}
+					}
+				}
+			}
+
+			return common.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'FILTER_FETCHED_SUCCESSFULLY',
+				result,
+			})
+		} catch (error) {
+			return error
+		}
+	}
+
+	static async getOrganizationIdBasedOnPolicy(userId) {
+		try {
+			let organization_ids = []
+
+			const menteeExtension = await menteeQueries.getMenteeExtension(userId, [
+				'external_mentor_visibility',
+				'organization_id',
+			])
+
+			if (menteeExtension.external_mentor_visibility === common.CURRENT) {
+				organization_ids.push(menteeExtension.organization_id)
+			} else if (menteeExtension.external_mentor_visibility === common.ASSOCIATED) {
+				let userOrgDetails = await userRequests.fetchDefaultOrgDetails(menteeExtension.organization_id)
+				if (userOrgDetails.success && userOrgDetails.data?.result?.related_orgs?.length > 0) {
+					organization_ids.push(...userOrgDetails.data.result.related_orgs, menteeExtension.organization_id)
+				}
+			} else if (menteeExtension.external_mentor_visibility === common.ALL) {
+				const organizationExtension = await organisationExtensionQueries.findAll(
+					{
+						external_mentor_visibility_policy: common.ALL,
+						mentor_visibility_policy: {
```

Comment:
Modify this query such that all all organisations with mentor_visibility_policy as ALL is fetched + all associated organisations of the current org is also fetched.